Author: DoloresPark <dolorespark@e-vertise.com>
Date:   Wed Mar 26 23:00:00 2014 -0400

    enable WiFi Display (Miracast) with audio disabled

diff --git a/overlay/frameworks/base/core/res/res/values/config.xml b/overlay/frameworks/base/core/res/res/values/config.xml
index fe8b8ab..ae9292d 100644
--- a/overlay/frameworks/base/core/res/res/values/config.xml
+++ b/overlay/frameworks/base/core/res/res/values/config.xml
@@ -239,4 +239,18 @@
     <!-- reference volume index for music stream to limit headphone volume and display warning -->
     <integer name="config_safe_media_volume_index">6</integer>
 
+    <!-- Whether WiFi display is supported by this device.
+         There are many prerequisites for this feature to work correctly.
+         Here are a few of them:
+         * The WiFi radio must support WiFi P2P.
+         * The WiFi radio must support concurrent connections to the WiFi display and
+           to an access point.
+         * The Audio Flinger audio_policy.conf file must specify a rule for the "r_submix"
+           remote submix module.  This module is used to record and stream system
+           audio output to the WiFi display encoder in the media server.
+         * The remote submix module "audio.r_submix.default" must be installed on the device.
+         * The device must be provisioned with HDCP keys (for protected content).
+    -->
+    <bool name="config_enableWifiDisplay">true</bool>
+
 </resources>
diff --git a/prebuilt/etc/audio_policy.conf b/prebuilt/etc/audio_policy.conf
index 9c91fe5..316af08 100644
--- a/prebuilt/etc/audio_policy.conf
+++ b/prebuilt/etc/audio_policy.conf
@@ -6,7 +6,12 @@
 # as well as the output device selected by default.
 # Devices are designated by a string that corresponds to the enum in audio.h
 
+# DoPa (20140326) - to enable audio for WiFi Display (Miracast) and possibly
+# generate a GIGANTIC number of error messages if it doesn't work, replace the
+# 'attached_input_devices' line below with:
+#    attached_input_devices AUDIO_DEVICE_IN_BUILTIN_MIC|AUDIO_DEVICE_IN_REMOTE_SUBMIX
+
 global_configuration {
   attached_output_devices AUDIO_DEVICE_OUT_SPEAKER
   default_output_device AUDIO_DEVICE_OUT_SPEAKER
   attached_input_devices AUDIO_DEVICE_IN_BUILTIN_MIC
@@ -84,5 +89,23 @@ audio_hw_modules {
       }
     }
   }
+  r_submix {
+    outputs {
+      submix {
+        sampling_rates 48000
+        channel_masks AUDIO_CHANNEL_OUT_STEREO
+        formats AUDIO_FORMAT_PCM_16_BIT
+        devices AUDIO_DEVICE_OUT_REMOTE_SUBMIX
+      }
+    }
+    inputs {
+      submix {
+        sampling_rates 48000
+        channel_masks AUDIO_CHANNEL_IN_STEREO
+        formats AUDIO_FORMAT_PCM_16_BIT
+        devices AUDIO_DEVICE_IN_REMOTE_SUBMIX
+      }
+    }
+  }
 
 }
