From: DoloresPark <dolorespark@e-vertise.com>
Date: Tue 15 Jul 2014 14:07:00 -0400
Subject: remove conflicting definition of 'off_t'

diff --git a/config.h b/config.h
index 651eb6b..2b6dc96 100644
--- a/config.h
+++ b/config.h
@@ -359,10 +359,7 @@
 #endif
 
 /* Define to `long int' if <sys/types.h> does not define. */
-#ifndef _OFF_T_DEFINED_
-#define _OFF_T_DEFINED_
-typedef long long off_t;
-#endif
+/* #undef off_t */
 
 /* Define to `unsigned int' if <sys/types.h> does not define. */
 /* #undef size_t */
