From: DoloresPark <dolorespark@e-vertise.com>
Date: Tue 15 Jul 2014 14:07:00 -0400
Subject: remove conflicting definition of 'off_t'

diff --git a/libexfat/exfat.h b/libexfat/exfat.h
index 269ba78..50f690c 100644
--- a/libexfat/exfat.h
+++ b/libexfat/exfat.h
@@ -24,11 +24,6 @@
 #ifndef EXFAT_H_INCLUDED
 #define EXFAT_H_INCLUDED
 
-#if defined(__ANDROID__)
-#define _OFF_T_DEFINED_
-typedef long long off_t;
-#endif
-
 #include <stdio.h>
 #include <stdlib.h>
 #include <time.h>
